spec: add a formal definition of the source language (#100)
nim-works/phy: spec/spec_old.md

## Phy Specification

> Note: this document is outdated an no longer authoritative. It's kept around

> > for some of it's ideas/directions that haven't been formalized yet.

Author: ```suggestion
> for some of its ideas/directions, which haven't all been formalized yet.
```
